app/models/miq_server.rb

```diff
@@ -474,7 +474,7 @@ def tenant_identity
   end
 
   def miq_region
-    ::MiqRegion.my_region
+    zone.miq_region
```

Comment:
Wow, interesting...

Reply:
I think you can just do `region_number`, right? You should be able to get the region right from the server objects `id`.

Reply:
Sorry, this is getting the object ... Yeah this will work I think Zone's miq_region method is using the id rather than the local region.

Reply:
To avoid bouncing through the zone and creating a second unnecessary query, you can do 

```ruby
MiqRegion.find_by(:region => my_region_number)
```

Reply:
👍